### ISSUE TITLE
Url shortening doesn't seem to work anymore in 9.8

### ISSUE THREAD

**Issue by archon810** [ShareX/ShareX#576]
I can't seem to shorten any url anymore. I have the shortener set to bit.ly, and it's been working for months/years, but not today.

I've tried Amazon, Verge, and other random urls, and they don't shorten. File upload is fine, gist is fine, but not bit.ly.

The logs just say this: 2015-03-29 16:11:29.739 - Hotkey triggered: Hotkey: Ctrl + Shift + Alt + Page Up, Description: Clipboard Upload, Job: ClipboardUpload

Any ideas?

**Comment by Jaex** [comment]
Unfortunately I can't reproduce it: http://bit.ly/1eB7oZ6

**Comment by archon810** [comment]
Nope, still happening. How can I help diagnose?

**Comment by Jaex** [comment]
Did you tried re authenticating to bitly? Also which URL you trying to shorten is not working?

**Comment by archon810** [comment]
That was the first thing I did. None of the urls are working, no matter
what I try.

**Comment by Jaex** [comment]
Maybe problem is not to do with bitly but instead your clipboard upload not even doing url shortening?
Did you checked this:
![](https://dl.dropboxusercontent.com/u/14076298/ShareX/2015/03/Tb5FCASukq.png)

**Comment by archon810** [comment]
Yes sir. http://i.imgur.com/3SrCsau.png.

That's the weird thing. This used to work for ages. I didn't change anything, then one day, poof.

**Comment by archon810** [comment]
Still not working in the latest version :(

**Comment by Jaex** [comment]
It works for everybody and without be able to reproduce it I can't know why it not works for you.
Is it works when you shorten using after upload task instead of clipboard upload? Is other url shorteners works?

**Comment by archon810** [comment]
Just tested after upload task - it worked. Shortened the imgur upload to my custom bit.ly one http://andp.lc/1PVCbQY.

**Comment by archon810** [comment]
Switching up the url shortener doesn't fix it. I even tried to log into goo.gl and switch to that, and still no reaction when pressing Ctrl+Shift+Alt+PageUp, which is my shortcut for uploading.

**Comment by Jaex** [comment]
http://i.imgur.com/3SrCsau.png in here you checked first one, how it supposed to shorten? But you didn't told it was instead uploading it too. So how that possible?

**Comment by archon810** [comment]
Sorry, I'm confused a bit. Can you explain/rephrase?

**Comment by Jaex** [comment]
In your config it will download image and re upload it because you checked first setting in that screenshot. So it can't shorten it. But you didn't told me it was uploading anything too so I was not suspicious of it.

**Comment by archon810** [comment]
I am not downloading/uploading images. Just random urls, like the one to this thread (https://github.com/ShareX/ShareX/issues/576). The only reason I tested an image just now was because you asked me to test what happens when I check shorten in "after upload task".

**Comment by Jaex** [comment]
When first option checked there it not running second option. This includes for third option too. All these 3 options URL jobs therefore if one of them triggered then other not runs.

**Comment by archon810** [comment]
This 100% definitely used to work. And I don't see why it shouldn't in this case. The first option is to download and upload if it contains a file URL. Which it doesn't. So it should move on to the next one, which is to shorten a url with a url shortener.

I understand your logic completely, but the keyword is "triggers." In case of google.com, the first rule does *not* trigger.

**Comment by Jaex** [comment]
File check happens in separate thread therefore it returns as soon as it is triggered.

**Comment by archon810** [comment]
Please reopen this, I don't agree with your resolution.

**Comment by archon810** [comment]
But don't you think that this is a bug, especially since it used to work? The fact that there's multithreading and race conditions involved means it's a bug or at least a feature that doesn't work 100%.

**Comment by Jaex** [comment]
It can't be used  to work, I don't remember changing its functionality. And It was kinda intentional why I did this because it will be inconsistent then. I mean if I want to shorten URLS then I would like to also shorten every URL not just non file ones. So I did this on purpose. Only one URL task can run actively for consistency. So it is not bug.

**Comment by archon810** [comment]
Then why is this not a radio button but a checkbox? A checkbox implies that all the given choices are optional and multiple can be checked and working?

As for the feature, I can't tell you how it used to work, but I haven't changed any settings and then suddenly after one of the updates, I couldn't shorten anything anymore. I don't understand why it can't just be a waterfall of consecutive if/thens? If first rule matches, do it and break the loop. If not, move on to the next one until something works or you're out of options.

**Comment by Jaex** [comment]
Reason is there was only one URL task before. Later I added two more and for not reset peoples setting not changed them to check boxes.

If that way easy like how you telling that if should work, I would do it. But there is threads involved therefore I can't just pass to other...

There is reason for everything. I thinked all of these before you ask these.

**Comment by McoreD** [comment]
Order of triggering should happen in the order of them being listed. If
not, this needs to be resolved.

**Comment by archon810** [comment]
@McoreD I've given up trying to make a case for fixing this and the fact that it's still buggy behavior to the user. I just unchecked the first box and won't be using the file url upload feature.

**Comment by McoreD** [comment]
Artem, can I suggest you to kindly create a new Task e.g. Ctrl+Shift+U to
specifically shorten URL and turn off everything else that's unnecessary or
conflicting?

**Comment by RadGH** [comment]
This looks unfortunate. I can't get URL shortening to work automatically, but I can if I manually create a short URL. I've followed the tips here to no avail.

Here are my settings:

https://dl.dropboxusercontent.com/s/18sudrj3koo8eau/2015-12-01_10-15-38.jpg?dl=0
https://dl.dropboxusercontent.com/s/i1v50bhgrdsmxgf/2015-12-01_10-16-52.jpg?dl=0
https://dl.dropboxusercontent.com/s/tje460juovcltrr/2015-12-01_10-17-06.jpg?dl=0
https://dl.dropboxusercontent.com/s/e2ayrcl676ura1f/2015-12-01_10-17-44.jpg?dl=0
https://dl.dropboxusercontent.com/s/jcgzgpui8thj7ip/2015-12-01_10-21-02.jpg?dl=0

Dropbox and Google (goo.gl) are authenticated. Google uses a Google Apps account if that matters, but I doubt it does.

The hotkey settings are defaults, except I changed Print Screen to do a region capture. The actual settings of each is the same.

When I take a screenshot, the following happens:

1. Image opens in local editor (I've tried disabling this, but that doesn't fix it).
2. Image saves to a folder
3. Image gets uploaded to dropbox
4. URL goes to clipboard for dropbox

URL never gets shortened. I've tried multiple providers. I think it's just skipping the step.

Here is the task list, the one "Shorten URL" was me manually creating a short URL. This worked. It just isn't automatic.

https://dl.dropboxusercontent.com/s/yf29ae6k7thg67l/2015-12-01_10-19-48.jpg?dl=0

Also note, it's not a problem that the short URL isn't being copied. It isn't being generated in the first place. Eg,

https://dl.dropboxusercontent.com/s/tgql30v743xy48p/2015-12-01_10-21-26.jpg?dl=0

**Comment by Jaex** [comment]
Google URL shortener not allows to shorten Dropbox urls, even in their web site.

**Comment by RadGH** [comment]
Ah crap. But it worked when I did it manually..? Wait... it didn't work then either!

http://tinyurl.com/pvleqgw

The task for shortening the URL was green. I figured that meant it worked. But I can't copy the short link from that task, only the original. I think that should have said it failed. Any sort of notice that the provider failed would have helped me figure that out. Maybe add a warning or a popup to explain the shortener failed?

Oh well. Doesn't really matter. Now that I got it figured out I'm good to go.

Sorry for the trouble, I didn't know :)
